cpp/CMakeLists.txt

```diff
@@ -220,6 +220,9 @@ if(BUILD_CUML_TESTS OR BUILD_PRIMS_TESTS)
   find_package(Threads)
 endif()
 
+# thrust before rmm, rmm before raft so we get the right version of thrust/rmm
+include(cmake/thirdparty/get_thrust.cmake)
+include(cmake/thirdparty/get_rmm.cmake)
```

Comment:
Should we have a flag or something to enable this conditionally and then be able to do these test PRs without adding the get_dependency.cmake files every time>?

Reply:
I discussed with @robertmaynard about this, since I was missing some historical context here. It sounds like this was tried before in #4643 which was closed without much discussion. I think I would argue that cuML _should_ explicitly depend on thrust and rmm. As I understand it, the only reason _not_ to do so (and to instead get those dependencies via raft) was because of patches that raft needed on thrust/rmm. However, it doesn't look like raft has patches for thrust/rmm at present (see raft's [patches directory](https://github.com/rapidsai/raft/tree/branch-23.06/cpp/cmake/patches)).

Reply:
Other context from @robertmaynard:
The original PR that added thrust and rmm to cuml: https://github.com/rapidsai/cuml/pull/3844/
PR that yanked thrust/rmm/etc in 22.02: https://github.com/rapidsai/cuml/pull/4256